lapixdl/convert.py

```diff
@@ -300,6 +307,136 @@ def __generate_coco_od_file(
     return coco_od
 
 
+# -----------------------------------------------------------------------
+# Functions to work with lapix DataFrame
+def lapix_to_annotations(lapix_df: LapixDataFrame) -> list[Annotation]:
+    return [
+        Annotation(
+            row['geometry'],
+            row['category_id'],
+            row['iscrowd'] if 'iscrowd' in row else 0
+        )
+        for _, row in lapix_df.iterrows()
+    ]
+
+
+# -----------------------------------------------------------------------
+# Functions to generate masks
+def sort_annotations_to_draw(
+    annotations: list[Annotation],
+    draw_order: tuple[int, ...] | None = None
+) -> list[Annotation]:
+    '''Will sort the annotations based on the draw_order, if draw_order
+    is None will sort in ascending order based in category value'''
+    items = defaultdict(list)
+
+    for ann in annotations:
+        items[ann.category_id].append(ann)
+
+    if draw_order is None:
+        draw_order = tuple(np.unique(list(items)))
+    elif not isinstance(draw_order, tuple):
+        raise ValueError('Unexpected value for `draw_order`. Need to be a tuple of int, or None.')
+
+    out = [ann for cat in draw_order for ann in items[cat]]
+
+    return out
+
+
+def draw_annotation(
+    target: Image.Image,
+    annotation: Annotation,
+    value: int | tuple[int, int, int],
+) -> Image.Image:
+    '''Image: Draw an Annotation into the target (an Image)'''
+    for geo in annotation:
+        pol_x, pol_y = geo.exterior.coords.xy
+        coords = list(zip(pol_x, pol_y))
+        ImageDraw.Draw(target).polygon(coords, fill=value)
+
+    return target
+
+
+def annotations_to_mask(
+    annotations: list[Annotation],
+    width: int = 1600,
+    height: int = 1200,
+) -> Mask:
+    """Mask: Generate a Mask with in the shape (height, width) with the
+    annotations"""
+    shape = (height, width)
+
+    annotations_sorted = sort_annotations_to_draw(annotations)
+
+    out = Image.fromarray(np.zeros(shape, dtype=np.uint8))
+    for ann in annotations_sorted:
+        out = draw_annotation(out, ann, ann.category_id)
+
+    return Mask(np.array(out))
+
+
+def __lapix_to_masks(
+    lapix_df: LapixDataFrame,
+    output_directory: str,
+    mask_extension: str = '.png'
+) -> None:
+    df_groupped = lapix_df.groupby('image_id')
+
+    for _, df_by_img in df_groupped:
+        annotations = lapix_to_annotations(df_by_img)
+
+        w = df_by_img['image_width'].unique()[0]
+        h = df_by_img['image_height'].unique()[0]
+
+        mask = annotations_to_mask(annotations, int(w), int(h))
+
+        img_name = basename(df_by_img.iloc[0]['image_name']) + mask_extension
+        out_path = os.path.join(output_directory, img_name)
+        mask.save(out_path)
+
+
+def lapix_to_masks(
```

Comment:
```suggestion
def save_lapixdf_as_masks(
```
It's not really a conversion method. Maybe we should put it inside the LapixDataset as `save_as_masks`?

Reply:
I don't know, because this function that ends up converting the dataset to semantic segmentation

Reply:
In this case, it can return a numpy mask iterator that can be used by the method that will save these masks to files.

Reply:
If we have a way to draw the polygons directly at NumPy arrays (without need to convert to PIL then to Numpy back), we can go in this way

Reply:
When I write this originally, I use the scikit-image option ([skimage.draw.polygon](https://scikit-image.org/docs/stable/api/skimage.draw.html#skimage.draw.polygon)), but add scikit as dependency just for it, I don't think make sense now

Reply:
Doesn't the `annotations_to_mask` method already returns the mask as a numpy array? I think this process can be written this way:

```
def lapix_to_masks(...):
  ...
  yield annotations_to_mask(annotations, int(w), int(h), draw_order)

def __save_masks_as_files(...params):
  for mask in lapix_to_masks(...params):
    img_name = basename(df_by_img.iloc[0]['image_name']) + mask_extension
    out_path = os.path.join(output_directory, img_name)
    mask.save(out_path)
```

This way you can also expose the `lapix_to_masks` converter.

Reply:
And yes, I've also searched about "drawing" forms in numpy arrays before and also found nothing 😞

Reply:
😞

Reply:
Moved to iterator at 1d9e449ac4a914c9ec2b1ae42da7fd2d58904c49